src/APIBasedAuth.ts

```diff
@@ -0,0 +1,153 @@
+import axios, { AxiosError, AxiosInstance, AxiosResponse } from 'axios';
+
+// exporting just for jest test coverage
+export const DEFAULT_BASE_URL = 'https://apps.us.lifeomic.com/auth/v1/api';
+
+// exporting just for jest test coverage
+export const formatAxiosError = <ErrorResponse = any>(
+  error: AxiosError<ErrorResponse> | unknown
+) => {
+  if (!axios.isAxiosError(error)) {
+    throw new AxiosError(String(error));
+  }
+  throw error.response?.data;
+};
+
+class APIBasedAuth {
+  private client: AxiosInstance;
+  private clientId: APIBasedAuth.Config['clientId'];
+  private session?: APIBasedAuth.Session;
+
+  constructor({ clientId, ...options }: APIBasedAuth.Config) {
+    if (!clientId) {
+      throw new Error('APIBasedAuth param clientId is required');
+    }
+
+    this.client = axios.create({
+      ...options,
+      baseURL: options.baseURL || DEFAULT_BASE_URL
+    });
+
+    this.clientId = clientId;
+  }
+
+  public confirmPasswordlessAuth(
+    baseInput: Omit<APIBasedAuth.VerifyPasswordlessAuthData, 'clientId'>
+  ) {
+    const input = {
+      ...baseInput,
+      // if session or username is not present get stored properties
+      ...(!baseInput.session || !baseInput.username ? this.session : null)
+    };
+    return this.client
```

Comment:
I see here an other places a lot of `.then` and `.catch` are used. Is that intentional rather than use async/await ?